### PR TITLE
fix 'unknown sources' setting visibility when opened from dialog

### PR DESCRIPTION
**Pull request by tamland**
alternative to #9821

### CODE REVIEW

**Comment by tamland** [comment]
@Montellese this ok with you?

File: xbmc/settings/windows/GUIWindowSettingsCategory.cpp [hunk 1]
```diff
@@ -201,15 +202,32 @@ void CGUIWindowSettingsCategory::Save()
...
+      SET_CONTROL_FOCUS(CONTROL_SETTINGS_START_BUTTONS + i, 0);
+      return;
+    }
+    for (const auto& group: m_categories[i]->GetGroups())
+    {
+      for (const auto& setting : group->GetSettings())
+      {
+        if (setting->GetId() == elementId)
+        {
+          SET_CONTROL_FOCUS(CONTROL_SETTINGS_START_BUTTONS + i, 0);
```
Comment (Montellese) [inline xbmc/settings/windows/GUIWindowSettingsCategory.cpp]: I'm pretty sure this should be `CONTROL_SETTINGS_START_CONTROL`

**Comment by Montellese** [comment]
Not a fan of the indexed for loop but no simple way around it. Apart from that it looks good.

**Comment by stefansaraev** [comment]
jenkins build this please

**Pull request merged**